#! /bin/sh /usr/share/dpatch/dpatch-run
## examples_compile_instructions by <peter@p12n.org>
##
## DP: Fix build instructions for /usr/share/doc/libsvn-doc/examples/*.c.

@DPATCH@
--- a/tools/examples/getlocks_test.c
+++ b/tools/examples/getlocks_test.c
@@ -15,13 +15,11 @@
  * history and logs, available at http://subversion.tigris.org/.
  * ====================================================================
  *
- *  To compile on unix against Subversion and APR libraries, try
- *  something like:
+ *  To compile on Debian:
  *
  *  cc getlocks_test.c -o getlocks_test \
- *  -I/usr/local/include/subversion-1 -I/usr/local/apache2/include \
- *  -L/usr/local/apache2/lib -L/usr/local/lib \
- *  -lsvn_client-1 -lsvn_ra-1 -lsvn_subr-1 -lapr-0 -laprutil-0
+ *  -I/usr/include/subversion-1 $(pkg-config --cflags apr-1) \
+ *  -lsvn_client-1 -lsvn_ra-1 -lsvn_subr-1
  *
  */
 
--- a/tools/examples/headrev.c
+++ b/tools/examples/headrev.c
@@ -15,13 +15,11 @@
  * history and logs, available at http://subversion.tigris.org/.
  * ====================================================================
  *
- *  To compile on unix against Subversion and APR libraries, try
- *  something like:
+ *  To compile on Debian:
  *
  *  cc headrev.c -o headrev \
- *  -I/usr/local/include/subversion-1 -I/usr/local/apache2/include \
- *  -L/usr/local/apache2/lib -L/usr/local/lib \
- *  -lsvn_client-1 -lsvn_ra-1 -lsvn_subr-1 -lapr-0 -laprutil-0
+ *  -I/usr/include/subversion-1 $(pkg-config --cflags apr-1) \
+ *  -lsvn_client-1 -lsvn_ra-1 -lsvn_subr-1
  *
  */
 
--- a/tools/examples/minimal_client.c
+++ b/tools/examples/minimal_client.c
@@ -25,12 +25,10 @@
  *  For a much more complex example, the svn cmdline client might be
  *  considered the 'reference implementation'.
  *
- *  From a Linux system, a typical commandline compile might look like:
+ *  To compile on Debian:
  *
  *  cc minimal_client.c -o minimal_client \
- *  -I/usr/local/include/subversion-1 -I/usr/local/apache2/include \
- *  -L/usr/local/apache2/lib -L/usr/local/lib \
- *  -lsvn_client-1 -lapr-0 -laprutil-0
+ *  -I/usr/include/subversion-1 $(pkg-config --cflags apr-1) -lsvn_client-1
  *
  */
 
--- a/tools/examples/svnput.c
+++ b/tools/examples/svnput.c
@@ -28,13 +28,10 @@
  * history and logs, available at http://subversion.tigris.org/.
  * ====================================================================
  *
- *  To compile on unix against Subversion and APR libraries, try
- *  something like:
+ *  To compile on Debian:
  *
  *  cc svnput.c -o svnput \
- *  -I/usr/local/include/subversion-1 -I/usr/local/apache2/include \
- *  -L/usr/local/apache2/lib -L/usr/local/lib \
- *  -lsvn_client-1 -lapr-0 -laprutil-0
+ *  -I/usr/include/subversion-1 $(pkg-config --cflags apr-1) -lsvn_client-1
  *
  */
 
--- a/tools/examples/testwrite.c
+++ b/tools/examples/testwrite.c
@@ -15,13 +15,11 @@
  * history and logs, available at http://subversion.tigris.org/.
  * ====================================================================
  *
- *  To compile on unix against Subversion and APR libraries, try
- *  something like:
+ *  To compile on Debian:
  *
  *  cc testwrite.c -o testwrite \
- *  -I/usr/local/include/subversion-1 -I/usr/local/apache2/include \
- *  -L/usr/local/apache2/lib -L/usr/local/lib \
- *  -lsvn_client-1 -lsvn_ra-1 -lsvn_subr-1 -lsvn-fs-1 -lapr-0 -laprutil-0
+ *  -I/usr/include/subversion-1 $(pkg-config --cflags apr-1) \
+ *  -lsvn_client-1 -lsvn_ra-1 -lsvn_subr-1 -lsvn_fs-1
  *
  */
 
